src/wallet/rpcwallet.cpp

```diff
@@ -2927,7 +2927,7 @@ static UniValue listunspent(const JSONRPCRequest& request)
         CTxDestination address;
         const CScript& scriptPubKey = out.tx->tx->vout[out.i].scriptPubKey;
         bool fValidAddress = ExtractDestination(scriptPubKey, address);
-        bool reused = avoid_reuse && pwallet->IsUsedDestination(address);
+        bool reused = avoid_reuse && pwallet->IsUsedDestination(out.tx->GetHash(), out.i);
```

Comment:
nit, you could move this line after L2932 - no point in calling `IsUsedDestination` if this address doesn't matter.